execution/scan/subquery.go

```diff
@@ -116,10 +116,10 @@ func (o *subqueryOperator) Next(ctx context.Context) ([]model.StepVector, error)
 			for j, vector := range vectors {
 				if vector.T > maxt {
 					o.lastVectors = vectors
+					o.lastCollected = j - 1
```

Comment:
if first vector has larger timestamp and we had no overflow before then lastCollected would be length of previous vectors and new vectors might be shorter resulting in a panic; if we set it like this its always set properly.

Reply:
first step vector of the current batch i mean